pkg/20.10.9/20.10.9.diff

```diff
@@ -0,0 +1,164 @@
+--- 20.10.9.orig.sh	2021-11-23 17:00:12.289617337 +0530
++++ 20.10.9.sh	2021-11-23 17:07:23.188430165 +0530
+@@ -24,25 +24,34 @@
+ # strip "v" prefix if present
+ VERSION="${VERSION#v}"
+ 
+-# The channel to install from:
+-#   * nightly
+-#   * test
+-#   * stable
+-#   * edge (deprecated)
+-DEFAULT_CHANNEL_VALUE="stable"
+-if [ -z "$CHANNEL" ]; then
+-	CHANNEL=$DEFAULT_CHANNEL_VALUE
+-fi
+-
+-DEFAULT_DOWNLOAD_URL="https://download.docker.com"
+-if [ -z "$DOWNLOAD_URL" ]; then
+-	DOWNLOAD_URL=$DEFAULT_DOWNLOAD_URL
+-fi
+-
+-DEFAULT_REPO_FILE="docker-ce.repo"
+-if [ -z "$REPO_FILE" ]; then
+-	REPO_FILE="$DEFAULT_REPO_FILE"
+-fi
++CHANNEL="stable"
++DOWNLOAD_URL="https://download.docker.com"
++REPO_FILE="docker-ce.repo"
++VERSION="20.10.9"
++DIND_TEST_WAIT=${DIND_TEST_WAIT:-3s}  # Wait time until docker start at dind test env
++
++# Issue https://github.com/rancher/rancher/issues/29246
++adjust_repo_releasever() {
++	DOWNLOAD_URL="https://download.docker.com"
++	case $1 in
++	7*)
++		releasever=7
++		;;
++	8*)
++		releasever=8
++		;;
++	*)
++		# fedora, or unsupported
++		return
++		;;
++	esac
++
++	for channel in "stable" "test" "nightly"; do
++		$sh_c "$config_manager --setopt=docker-ce-${channel}.baseurl=${DOWNLOAD_URL}/linux/centos/${releasever}/\\\$basearch/${channel} --save";
++		$sh_c "$config_manager --setopt=docker-ce-${channel}-debuginfo.baseurl=${DOWNLOAD_URL}/linux/centos/${releasever}/debug-\\\$basearch/${channel} --save";
++		$sh_c "$config_manager --setopt=docker-ce-${channel}-source.baseurl=${DOWNLOAD_URL}/linux/centos/${releasever}/source/${channel} --save";
++	done
++}
+ 
+ mirror=''
+ DRY_RUN=${DRY_RUN:-}
+@@ -71,6 +80,18 @@
+ 		;;
+ esac
+ 
++start_docker() {
++	if [ ! -z $DIND_TEST ]; then
++		# Starting dockerd manually due to dind env is not using systemd
++		dockerd &
++		sleep $DIND_TEST_WAIT
++	elif [ -d '/run/systemd/system' ] ; then
++		$sh_c 'systemctl start docker'
++	else
++		$sh_c 'service docker start'
++	fi
++}
++
+ command_exists() {
+ 	command -v "$@" > /dev/null 2>&1
+ }
+@@ -352,6 +373,12 @@
+ 				dist_version="$(. /etc/os-release && echo "$VERSION_ID")"
+ 			fi
+ 		;;
++		
++		oracleserver|ol)
++			lsb_dist="ol"
++			# need to switch lsb_dist to match yum repo URL
++			dist_version="$(rpm -q --whatprovides redhat-release --queryformat "%{VERSION}\n" | sed 's/\/.*//' | sed 's/\..*//' | sed 's/Server*//')"
++		;;
+ 
+ 		*)
+ 			if command_exists lsb_release; then
+@@ -448,16 +475,18 @@
+ 					# Install docker-ce-rootless-extras without "--no-install-recommends", so as to install slirp4netns when available
```

Comment:
In the new script there another addition. You can find it [here](https://github.com/rayandas/install-docker/blob/132bde040882475c1f4a25978181604ae18ee654/pkg/20.10.9/20.10.9.diff?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L96). 
Better to wait till [20.10.8 PR](https://github.com/rancher/install-docker/pull/83) to get merged. then you can follow the process again with `PREVIOUS_ADD_DOCKER_VERSION=20.10.8 ADD_DOCKER_VERSION=20.10.9 make add-new-version` and then `make generate`. It will be fixed also the additional changes will get added.